### PR TITLE
server: make server testable with client

### PR DESCRIPTION
**Pull request by zojw**
tiny fix to pass #781 client example test

### CODE REVIEW

**Comment by tisonkun** [comment]
@zojw can you rebase this patch onto latest main branch?

**Comment by zojw** [comment]
> @zojw can you rebase this patch onto latest main branch?

done~

**Review by tisonkun — APPROVED** [review]
Verified that we can run a minimal KV put/get example. +1 to merge.

May I ask what the change you made in this patch to fix the issue? It's not quite straightforward from the diff itself.

**Review by w41ter — APPROVED** [review]

File: src/server/src/node/job/report_state.rs [hunk 1]
```diff
@@ -114,6 +114,7 @@ async fn report_state_updates(roots: &mut Vec<String>, request: ReportRequest) {
 /// If one day you find that reporting has become a bottleneck, you can consider optimizing this
 /// code.
 async fn issue_report_request(addr: String, request: &ReportRequest) -> Result<()> {
+    let addr = format!("http://{}", addr);
```
Comment (tisonkun) [inline src/server/src/node/job/report_state.rs:117]: Weird. I think that RootClient and effectively NodeClient should prepend `"http://"`.

```rust
// RootClient
pub async fn connect(addr: String) -> Result<Self, crate::Error> {
    let node_client = NodeClient::connect(addr).await?;
    // ...
}

// NodeClient
pub async fn connect(addr: String) -> Result<Self, tonic::transport::Error> {
    let addr = format!("http://{}", addr);
    // ...
}
```

Comment (zojw) [reply src/server/src/node/job/report_state.rs:117]: yes, `engula_client::RootClient` can handle it rightly, but these methods use protobuf's RootClient and self-maintain root address list and miss it

Comment (tisonkun) [reply src/server/src/node/job/report_state.rs:117]: Well. This `RootClient` is from protobuf generated :P

**Pull request merged**